app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogDateResarvationView.kt

```diff
@@ -1,14 +1,35 @@
 package com.example.parkingsystemkotlin.mvp.view
 
 import androidx.fragment.app.DialogFragment
+import com.example.parkingsystemkotlin.databinding.FragmentDialogDatePickerBinding
+import com.example.parkingsystemkotlin.listener.ListenerDialogFragmentDate
 import com.example.parkingsystemkotlin.mvp.contract.DialogDateResarvationContract
 import com.example.parkingsystemkotlin.mvp.view.base.FragmentView
+import java.util.GregorianCalendar
+import java.util.Calendar
 
 class DialogDateResarvationView(
     fragment: DialogFragment,
+    private val binding: FragmentDialogDatePickerBinding
 ) : FragmentView(fragment), DialogDateResarvationContract.DialogFragmentDateReservationView {
 
     override fun dismissPicker() {
         (fragment as DialogFragment).dismiss()
     }
+
+    override fun setDateTime(listenerDate: ListenerDialogFragmentDate, startDatePicker: Boolean) {
+        listenerDate.setDateTime(getCalendarDateTime(), startDatePicker)
+    }
+
+    private fun getCalendarDateTime(): Calendar {
+        val extractDateTime: Calendar = GregorianCalendar()
+        extractDateTime.set(
+            binding.datePickerDialogFragment.year,
+            binding.datePickerDialogFragment.month,
+            binding.datePickerDialogFragment.dayOfMonth,
+            binding.timePickerDialogFragment.hour,
+            binding.timePickerDialogFragment.minute
+        )
+        return extractDateTime
```

Comment:
here you can avoid using return and creating a new var using  = GregorianCalendar().set( ... )

Reply:
if I do this the type is changed from calendar and it becomes until